module/journal.js

```diff
@@ -396,6 +412,23 @@ export default class WorldAnvilBrowser extends Application {
 
   /* -------------------------------------------- */
 
+  /**
+   * Calls ._calculateCategoryVisibility
+   * On first call will init ._collapsedCategories before calling it.
+   */
+  _refreshCategoryVisibility() {
+    if(this._firstInit) {
+      const firstLevelsIds = this.tree.children.map( c => c.id );
+      firstLevelsIds.push(CATEGORY_ID.root);
+
+      this._collapsedCategories = [...this.categories.values()]
+        .map( cat => cat.id )
+        .filter(catId => !firstLevelsIds.includes(catId));
```

Comment:
This iterates 3 times, once to deconstruct the array from the `.values()` iterator, and then again twice more with `.map()` and `.filter()`. 

Suggest instead using a pattern like:
```js
this._collapsedCategories = [];
for ( const category of this.categories.values() ) {
  if ( firstLevelsIds.includes(category.id ) ) this._collapsedCategories.push(category.id);
}
```
This only iterates once.